### PR TITLE
ip_isValidPassword - String extension for validating password against requirements

### PR DESCRIPTION
**Pull request by petey3**
Pass in array of `PasswordRequirement` values and use regex to check that the string passes those tests. `PasswordRequirement` values include the inclusion/exclusion of at least one:
- Special Character
- Number
- Uppercase
- Lowercase

as well as a Minimum and Maximum length for the password.

### CODE REVIEW

File: SwiftWisdom/Core/StandardLibrary/String/String+EmailValidation.swift [hunk 1]
```diff
@@ -19,4 +60,19 @@ extension String {
...
+    func ip_isValidPassword(passwordRequirements: [PasswordRequirement]) -> Bool {
+        for requirement in passwordRequirements {
+            let found = rangeOfString(requirement.expression, options: .RegularExpressionSearch) != nil
+            if (!requirement.exclude && !found) || (requirement.exclude && found) {
+                return false
+            }
+        }
+        return true
+    }
+}
```
Comment (Benuuu) [inline SwiftWisdom/Core/StandardLibrary/String/String+EmailValidation.swift]: newline at eof

File: SwiftWisdom/Core/StandardLibrary/String/String+EmailValidation.swift [hunk 2]
```diff
@@ -19,4 +60,19 @@ extension String {
     public func ip_isValidEmail() -> Bool {
         return rangeOfString(".@.+\\..", options: .RegularExpressionSearch) != nil
     }
-}
+
+
+    /**
+     Validate if an email at least conforms to an array of requirements. For character types, requirments can define if they should be excluded or appear at least once. Minimum and maximum length are also requirment types. Does not define any expectation beyond what is in the array of requirements (i.e. this function will always return true if the password requirement array is empty)
```
Comment (Benuuu) [inline SwiftWisdom/Core/StandardLibrary/String/String+EmailValidation.swift]: `if a password`

Comment (Benuuu) [inline SwiftWisdom/Core/StandardLibrary/String/String+EmailValidation.swift]: This line is really long, can you keep it at a 120 character line limit?

Comment (petey3) [inline SwiftWisdom/Core/StandardLibrary/String/String+EmailValidation.swift]: yessir

**Comment by paulrolfe** [comment]
@petey3 rebase and :shipit:

This has been waiting for a while, it looks like.

**Pull request merged**